agent/fw_drupal8.c

```diff
@@ -605,6 +605,89 @@ NR_PHP_WRAPPER(nr_drupal94_invoke_all_with_clean) {
 NR_PHP_WRAPPER_END
 #endif  // OAPI
 
+/*
+ * Purpose: Instrument Drupal Attribute Hooks for Drupal 11.1+
+ *
+ * Params: 1. A zval pointer to the moduleHandler instance in use by Drupal.
+ *
+ * Return: bool
+ *
+ */
+static bool nr_drupal_hook_attribute_instrument(zval* module_handler) {
+  zval* hook_implementation_map = NULL;
+
+  nr_php_string_hash_key_t* hook_key = NULL;
+  zval* hook_val = NULL;
+  nr_php_string_hash_key_t* class_key = NULL;
+  zval* class_val = NULL;
+  nr_php_string_hash_key_t* method_key = NULL;
+  zval* module_val = NULL;
+
+  char* hookpath = NULL;
+
+  hook_implementation_map = nr_php_get_zval_object_property(
+      module_handler, "hookImplementationsMap");
+
+  if (NULL == hook_implementation_map) {
+    nrl_warning(NRL_FRAMEWORK, "NULL hookImplementationsMap object property");
+    return false;
+  }
+
+  if (!nr_php_is_zval_valid_array(hook_implementation_map)) {
+    nrl_warning(NRL_FRAMEWORK,
+                "hookImplementationsMap property not a valid array");
+    return false;
+  }
+
+  ZEND_HASH_FOREACH_STR_KEY_VAL(Z_ARRVAL_P(hook_implementation_map), hook_key,
+                                hook_val) {
+    if ((NULL == hook_key) || (0 == nr_php_is_zval_valid_array(hook_val))) {
+      nrl_warning(NRL_FRAMEWORK,
+                  "hookImplementationsMap[hook]: invalid key or value");
+      return false;
+    }
+
+    ZEND_HASH_FOREACH_STR_KEY_VAL(Z_ARRVAL_P(hook_val), class_key, class_val) {
+      if ((NULL == class_key) || (0 == nr_php_is_zval_valid_array(class_val))) {
+        nrl_warning(NRL_FRAMEWORK,
+                    "hookImplementationsMap[class]: invalid key or value");
+        return false;
```

Comment:
Similar to https://github.com/newrelic/newrelic-php-agent/pull/1030/files#r1974532294, do we want to exit the whole function or just skip this round of the loop.